addons/product/report/product_pricelist_report.py

```diff
@@ -17,14 +17,15 @@ def get_html(self, data):
         return self.env.ref('product.report_pricelist_page')._render(render_values)
 
     def _get_report_data(self, data, report_type='html'):
-        quantities = data['quantities'] or [1]
+        quantities = data.get('quantities', [1])
 
-        pricelist_id = data['pricelist_id'] and int(data['pricelist_id']) or None
+        data_pricelist_id = data.get('pricelist_id')
+        pricelist_id = int(data.get('pricelist_id')) if data_pricelist_id else None
```

Comment:
```suggestion
        pricelist_id = data_pricelist_id and int(data_pricelist_id)
```
little simplification, but not a problem if you wanna merge the current diff